Apps/WebClient/src/ClientApp/src/components/private/reports/RecommendationsDialogComponent.vue

```diff
@@ -0,0 +1,175 @@
+import { useReportStore } from "@/stores/report";
+
+interface Props {
+    hdid: string;
+    isDependent?: boolean;
+}
+const props = withDefaults(defineProps<Props>(), {
+    isDependent: false,
+});
+
+const reportFilter = ReportFilterBuilder.buildEmpty();
+
+defineExpose({ showDialog });
+
+const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+const reportStore = useReportStore();
+const errorStore = useErrorStore();
+
+const messageModal = ref<InstanceType<typeof MessageModalComponent>>();
+const recommendationsReportComponent =
+    ref<InstanceType<typeof ImmunizationReportComponent>>();
+
+const isVisible = ref(false);
+const isGeneratingReport = ref(false);
+const hasRecords = ref(false);
+const reportFormatType = ref<ReportFormatType>(ReportFormatType.PDF);
+
+function showConfirmationModal(type: ReportFormatType): void {
+    reportFormatType.value = type;
+    messageModal.value?.showModal();
+}
+
+function downloadReport() {
+    isGeneratingReport.value = true;
+    recommendationsReportComponent.value
+        ?.generateReport(
+            reportFormatType.value,
+            reportStore.getHeaderData(props.hdid, reportFilter)
+        )
+        .then((result: RequestResult<Report>) => {
+            const mimeType =
+                reportMimeTypeMap.get(reportFormatType.value) ?? "";
+            const downloadLink = `data:${mimeType};base64,${result.resourcePayload.data}`;
+            fetch(downloadLink).then((res) =>
+                res
+                    .blob()
+                    .then((blob) =>
+                        saveAs(blob, result.resourcePayload.fileName)
+                    )
+            );
+        })
+        .catch((err: ResultError) => {
+            logger.error(err.resultMessage);
+            if (err.statusCode === 429) {
+                errorStore.setTooManyRequestsError("recommendationsDialog");
+            } else {
+                errorStore.addError(
+                    ErrorType.Download,
+                    ErrorSourceType.ExportRecords,
+                    err.traceId
+                );
+            }
+        })
+        .finally(() => {
+            isGeneratingReport.value = false;
+        });
+}
+
+function showDialog() {
+    isVisible.value = true;
+}
+</script>
+
+<template>
+    <div class="d-flex justify-content">
+        <v-dialog v-model="isVisible" max-width="1000px" persistent>
+            <v-card data-testid="recommendations-dialog">
+                <v-card-title>
+                    <v-toolbar
+                        title="Vaccine Recommendations"
+                        color="transparent"
+                        ><HgIconButtonComponent
+                            id="recommendations-dialog-close-button"
+                            data-testid="recommendations-dialog-close-button"
+                            icon="close"
+                            @click="isVisible = false"
+                        />
+                    </v-toolbar>
```

Comment:
```suggestion
                    <v-toolbar title="Vaccine Recommendations" color="transparent">
                        <HgIconButtonComponent
                            id="recommendations-dialog-close-button"
                            data-testid="recommendations-dialog-close-button"
                            icon="close"
                            @click="isVisible = false"
                        />
                    </v-toolbar>
```

May need to confirm linting is okay with this suggestion.

Reply:
sigh this auto complete can really cause some really bad markup structures.